src/framework/results/data/average_snapshot.hpp

```diff
@@ -108,23 +111,21 @@ void AverageSnapshot<T>::combine(AverageSnapshot<T> &&other) noexcept {
   other.clear();  
 }
 
-//------------------------------------------------------------------------------
-// JSON serialization
-//------------------------------------------------------------------------------
 template <typename T>
-void to_json(json_t &js, const AverageSnapshot<T> &snapshot) {
-  js = json_t::object();
-  for (const auto &outer_pair : snapshot.data()) {
-    for (const auto &inner_pair : outer_pair.second) {
+json_t AverageSnapshot<T>::to_json() {
+  json_t js = json_t::object();
+  for (auto &outer_pair : data_) {
+    for (auto &inner_pair : outer_pair.second) {
       // Store mean and variance for snapshot
-      json_t datum = inner_pair.second;
+      json_t datum = inner_pair.second.to_json();
       // Add memory key if there are classical registers
       auto memory = inner_pair.first;
```

Comment:
I know is not on this PR, but do we need to copy that string? Can't we just take a reference?